File: components/albums/src/main/java/org/superbiz/moviefun/albums/AlbumsController.java
package org.superbiz.moviefun.albums;

import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/albums")
public class AlbumsController {
    private AlbumsRepository albumsRepository;

    public AlbumsController(AlbumsRepository albumsRepository) {
        this.albumsRepository = albumsRepository;
    }

    @PostMapping
    public void addAlbum(@RequestBody Album album) {
        this.albumsRepository.addAlbum(album);
    }

    @GetMapping("/{id}")
    public Album find(@PathVariable("id") Long id) {
        return this.albumsRepository.find(id);
    }

    @GetMapping
    public List<Album> getAlbums() {
        return this.albumsRepository.getAlbums();
    }

    @DeleteMapping
    public void deleteAlbum(Album album) {
        this.albumsRepository.deleteAlbum(album);
    }

    @PutMapping
    public void updateAlbum(Album album) {
        this.albumsRepository.updateAlbum(album);
    }
}

File: components/albums/src/main/java/org/superbiz/moviefun/albums/AlbumsController.kt
package org.superbiz.moviefun.albums

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/albums")
class AlbumsController(private val albumsRepository: AlbumsRepository) {

    @PostMapping
    fun addAlbum(@RequestBody album: Album) = albumsRepository.addAlbum(album)

    @GetMapping
    fun index(): List<Album> =  albumsRepository.getAlbums()

    @GetMapping("/{albumId}")
    fun details(@PathVariable albumId: Long): Album =  albumsRepository.find(albumId)
}
